sycl/plugins/unified_runtime/ur/adapters/level_zero/usm.hpp

```diff
@@ -40,75 +40,100 @@ class UsmAllocationException {
   ur_result_t getError() const { return Error; }
 };
 
-// Implements memory allocation via L0 RT for USM allocator interface.
-class USMMemoryProvider {
-private:
+// UMF memory provider interface for USM.
+class USMMemoryProviderBase {
+protected:
+  ur_context_handle_t Context;
+  ur_device_handle_t Device;
+
   ur_result_t &getLastStatusRef() {
     static thread_local ur_result_t LastStatus = UR_RESULT_SUCCESS;
     return LastStatus;
   }
 
-protected:
-  ur_context_handle_t Context;
-  ur_device_handle_t Device;
-  size_t MinPageSize;
-
   // Internal allocation routine which must be implemented for each allocation
   // type
-  virtual ur_result_t allocateImpl(void **ResultPtr, size_t Size,
-                                   uint32_t Alignment) = 0;
+  virtual ur_result_t allocateImpl(void **, size_t, uint32_t) = 0;
 
 public:
-  umf_result_t initialize(ur_context_handle_t Ctx, ur_device_handle_t Dev);
-  umf_result_t alloc(size_t Size, size_t Align, void **Ptr);
-  umf_result_t free(void *Ptr, size_t Size);
-  void get_last_native_error(const char **ErrMsg, int32_t *ErrCode);
-  umf_result_t get_min_page_size(void *, size_t *);
-  umf_result_t get_recommended_page_size(size_t, size_t *) {
+  virtual void get_last_native_error(const char **ErrMsg, int32_t *ErrCode) {
+    std::ignore = ErrMsg;
```

Comment:
This is unrelated to this PR, but I think it would be good to actually call `zeDriverGetLastErrorDescription` here.